### PR TITLE
Replace ack_only with retransmittable

### PR DESCRIPTION
**Pull request by ianswett**
Fixes the worst errors cited in #1932 

Feel free to make more suggestions.

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -657,9 +657,9 @@ are as follows:
 
 * packet_number: The packet number of the sent packet.
 
-* ack_only: A boolean that indicates whether a packet contains only
-  ACK or PADDING frame(s).  If true, it is still expected an ack will
-  be received for this packet, but it is not retransmittable.
+* retransmittable: A boolean that indicates whether a packet contains
+  frame(s) besides ACK or PADDING.  If true, it is expected an ACK will
+  be received, though the peer may delay it by up to the MaxAckDelay.
```
[the remark this answers is not in the document]
Comment (mikkelfj) [reply draft-ietf-quic-recovery.md]: Yes I am aware, but didn't find it important in relation to the text being discussed.

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -657,9 +657,9 @@ are as follows:
 
 * packet_number: The packet number of the sent packet.
 
-* ack_only: A boolean that indicates whether a packet contains only
-  ACK or PADDING frame(s).  If true, it is still expected an ack will
-  be received for this packet, but it is not retransmittable.
+* retransmittable: A boolean that indicates whether a packet contains
+  frame(s) besides ACK or PADDING.  If true, it is expected an ACK will
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
  frame(s) besides ACK or PADDING.  If true, it is expected that an
  acknowledgement will be received, though the peer may delay it by
  up to the MaxAckDelay.
```

File: draft-ietf-quic-recovery.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ianswett) [reply draft-ietf-quic-recovery.md]: Ok, I updated the text to sidestep the issue.

Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
```

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -616,12 +617,12 @@ sent_packets:
...
-  indicates the packet's size in bytes.  sent_packets is ordered by packet
-  number, and packets remain in sent_packets until acknowledged or lost.  A
-  sent_packets data structure is maintained per packet number space, and ACK
-  processing only applies to a single space.
+  was sent, a boolean indicating whether the packet is retransmittable,
+  a boolean indicating whether it counts towards bytes in flight, and a size
+  field indicating the packet's size in bytes.  sent_packets is ordered and
+  indexed by packet number. Packets remain in sent_packets until acknowledged
+  or lost.  A sent_packets data structure is maintained per packet number space,
+  and ACK processing only applies to a single space.
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
  and processing of an ACK frame only applies to a single space.
```

File: draft-ietf-quic-recovery.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikkelfj) [reply draft-ietf-quic-recovery.md]: Reverting a bit on this: the text gives the impression that ACK's happen eventually, but if the ACK frame grows fat, old packets will fall out. For example if sending massive amounts of padding for no good reason. I suppose this is covered in transport but ...

File: draft-ietf-quic-recovery.md [hunk 4]
```diff
@@ -131,8 +132,9 @@ mechanisms ensure that data and frames that need reliable delivery are
 acknowledged or declared lost and sent in new packets as necessary. The types
 of frames contained in a packet affect recovery and congestion control logic:
 
-* All packets are acknowledged, though packets that contain only ACK
-  and PADDING frames are not acknowledged immediately.
+* All packets are acknowledged, though packets that contain no
+  retransmittable frames are only acknowledged along with retransmittable
+  packets.
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: How about: "Packets containing only ACK or PADDING frames are not required to be acknowledged immediately by a receiver. They might be eventually included in acknowledgements when subsequent packets containing retransmittable frames are received by the receiver."

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: I like what I have in this PR currently better.  Is there something you don't like about the current text that you're trying to fix?

**Comment by ianswett** [comment]
I'm merging this because it's starting to overlap with #1939.  I'm happy to fix any outstanding issues in that PR.

**Pull request merged**